File: Prompt Engineering/L6_ACP_Choice_Board_Showcase.js
// ========================================================================
// Prompt Engineering - Lesson 6: Prompt-to-Project Showcase (Capstone)
// After Class Project (ACP): Choice Board Showcase
// Platform: Code.org Game Lab (JavaScript)
//
// INSTRUCTIONS:
// This file contains complete working solutions for all 5 Choice Board Themes:
// - Theme A: Fruit Catcher
// - Theme B: Traffic Dodge (Active below)
// - Theme C: Coral Explorer
// - Theme D: Jungle Runner
// - Theme E: Treasure Door
// ========================================================================

// ========================================================================
// THEME B: TRAFFIC DODGE (Active Implementation)
// ========================================================================

// Step 1: Background and Sprites
var bg = createSprite(200, 200);
bg.setAnimation("background_city_1");

// Player car
var car = createSprite(200, 340);
car.setAnimation("car_red_1");
car.scale = 0.6;

// Traffic cone hazard
var cone = createSprite(randomNumber(40, 360), -20);
cone.setAnimation("cone_1");
cone.scale = 0.6;

// Game state
var lives = 3;
var score = 0;
var timer = 30;

function draw() {
  background("skyblue");

  if (timer > 0 && lives > 0) {
    // Step 2 & 3: Movement & Hazards
    if (keyDown("left")) car.x -= 5;
    if (keyDown("right")) car.x += 5;
    car.x = constrain(car.x, 30, 370);

    // Hazard falls downward
    cone.y += 6;
    if (cone.y > 430) {
      cone.y = -20;
      cone.x = randomNumber(40, 360);
      score++; // Score increases for successfully dodged cones
    }

    // Step 4: Collision detection
    if (car.isTouching(cone)) {
      lives--;
      cone.y = -20;
      cone.x = randomNumber(40, 360);
    }

    // Timer (30 FPS)
    if (World.frameCount % 30 === 0) {
      timer--;
    }

    // Step 5: Draw sprites and HUD overlay
    drawSprites();

    fill("white");
    rect(0, 0, 400, 40);
    fill("black");
    textSize(18);
    text("Dodged: " + score, 20, 25);
    text("Lives: " + lives, 170, 25);
    text("Time: " + timer + "s", 300, 25);
  } else {
    // Step 6: Win / Lose End Screen
    drawSprites();
    fill("yellow");
    textSize(28);

    if (lives > 0 && timer <= 0) {
      text("You Win! 🏆", 140, 200);
      textSize(18);
      fill("white");
      text("You survived the highway dash!", 90, 240);
    } else {
      text("Game Over! 💥", 125, 200);
      textSize(18);
      fill("white");
      text("Car damaged! Try again!", 115, 240);
    }
  }
}

/*
// ========================================================================
// THEME C: CORAL EXPLORER (Reference Code)
// ========================================================================
var bg = createSprite(200, 200);
bg.setAnimation("background_underwater_11_1");
var diver = createSprite(60, 320);
diver.setAnimation("diver_1");
diver.scale = 0.35;
var pearl = createSprite(randomNumber(40, 360), randomNumber(60, 320));
pearl.setAnimation("pearl_1");
pearl.scale = 0.25;
var score = 0;
var timer = 30;

function drawCoralExplorer() {
  background("skyblue");
  if (timer > 0) {
    if (keyDown("left")) diver.x -= 4;
    if (keyDown("right")) diver.x += 4;
    if (keyDown("up")) diver.y -= 4;
    if (keyDown("down")) diver.y += 4;
    diver.x = constrain(diver.x, 20, 380);
    diver.y = constrain(diver.y, 60, 360);

    if (diver.isTouching(pearl)) {
      score++;
      pearl.x = randomNumber(40, 360);
      pearl.y = randomNumber(60, 320);
    }
    if (World.frameCount % 30 === 0) timer--;
    drawSprites();

    fill("white"); rect(0, 0, 400, 40);
    fill("black"); textSize(18);
    text("Pearls: " + score, 20, 25); text("Time: " + timer, 300, 25);
  } else {
    drawSprites();
    fill("yellow"); textSize(28);
    text(score >= 10 ? "You Win! 🤿" : "Game Over!", 130, 210);
  }
}
*/

/*
// ========================================================================
// THEME D: JUNGLE RUNNER (Reference Code)
// ========================================================================
var bg = createSprite(200, 200);
bg.setAnimation("background_jungle");
var runner = createSprite(80, 340);
runner.setAnimation("monkey");
runner.scale = 0.35;
var log = createSprite(420, 350);
log.setAnimation("log_1");
log.scale = 0.5;
var score = 0;
var lives = 3;
var timer = 30;

function drawJungleRunner() {
  background("skyblue");
  if (timer > 0 && lives > 0) {
    runner.x += 2;
    if (runner.x > 420) { runner.x = -20; score++; }

    if (keyWentDown("space") && runner.y >= 340) {
      runner.velocityY = -10;
    }
    runner.velocityY += 0.6;
    if (runner.y > 340) { runner.y = 340; runner.velocityY = 0; }

    log.x -= 4;
    if (log.x < -20) log.x = 420;

    if (runner.isTouching(log)) {
      lives--;
      log.x = 420;
    }
    if (World.frameCount % 30 === 0) timer--;
    drawSprites();

    fill("white"); rect(0, 0, 400, 40);
    fill("black"); textSize(18);
    text("Score: " + score, 20, 25); text("Lives: " + lives, 170, 25); text("Time: " + timer, 300, 25);
  } else {
    drawSprites();
    fill("yellow"); textSize(28);
    text(lives > 0 ? "Jungle Cleared! 🐒" : "Game Over!", 110, 210);
  }
}
*/

/*
// ========================================================================
// THEME E: TREASURE DOOR (Reference Code)
// ========================================================================
var bg = createSprite(200, 200);
bg.setAnimation("background_castle_1");
var door = createSprite(200, 220);
door.setAnimation("door_closed_1");
door.scale = 0.7;
var keyA = createSprite(110, 330);
keyA.setAnimation("key_gold_1");
var keyB = createSprite(200, 330); // Correct Key!
keyB.setAnimation("key_silver_1");
var keyC = createSprite(290, 330);
keyC.setAnimation("key_copper_1");
var won = false;

function drawTreasureDoor() {
  background("skyblue");
  drawSprites();
  fill("white"); textSize(16);
  text("Click the right key to open the treasure door!", 40, 40);

  if (mouseWentDown("leftButton")) {
    if (mousePressedOver(keyB)) {
      door.setAnimation("door_open_1");
      won = true;
    }
  }
  if (won) {
    fill("yellow"); textSize(26);
    text("Treasure Unlocked! 🗝️✨", 80, 150);
  }
}
*/

File: Prompt Engineering/L6_ACP_Choice_Board_Showcase.py
"""
Prompt Engineering - Lesson 6: Prompt-to-Project Showcase
After Class Project (ACP): Choice Board Showcase

Interactive Python terminal player supporting all 5 Choice Board game themes.
"""

import time
import random

def banner(title):
    print("\n" + "=" * 70)
    print(f"   🏆 {title.upper()} 🎮")
    print("=" * 70)

def play_traffic_dodge():
    banner("Theme B: Traffic Dodge Arcade")
    print("Goal: Dodge cones on the highway! Survive for 10 rounds with lives > 0.")
    print("Controls: 'a' (steer left), 'd' (steer right), Enter (stay in lane)\n")

    car_lane = 2
    lives = 3
    rounds = 10
    lanes = 5

    while rounds > 0 and lives > 0:
        cone_lane = random.randint(0, lanes - 1)

        print("\n" + "—" * 35)
        print(f"🚗 DODGES LEFT: {rounds} | LIVES: {'❤️' * lives}")
        print("—" * 35)

        road_hazard = ["·"] * lanes
        road_hazard[cone_lane] = "⚠️"
        print(f"Highway: {' '.join(road_hazard)}")

        road_car = ["·"] * lanes
        road_car[car_lane] = "🏎️"
        print(f"Car:     {' '.join(road_car)}")

        steer = input("\nSteer (a/d): ").strip().lower()
        if steer == 'a' and car_lane > 0:
            car_lane -= 1
        elif steer == 'd' and car_lane < lanes - 1:
            car_lane += 1

        if car_lane == cone_lane:
            lives -= 1
            print("💥 CRASH! You clipped a traffic cone! (-1 Life)")
        else:
            print("💨 Clean dodge! Speeding forward!")

        rounds -= 1
        time.sleep(0.3)

    print("\n" + "=" * 40)
    if lives > 0:
        print("🏆 YOU WIN! 🏁")
        print(f"You conquered the Traffic Dodge challenge with {lives} lives remaining!")
    else:
        print("💥 GAME OVER! Vehicle totaled!")
    print("=" * 40)

def play_treasure_door():
    banner("Theme E: Treasure Door Challenge")
    print("A locked stone portal stands before you in the castle.")
    print("Three antique keys lie on the pedestal:\n")
    print(" [A] 🔑 Golden Key")
    print(" [B] 🗝️ Silver Key")
    print(" [C] 🗝️ Copper Key\n")

    choice = input("Which key will you insert into the lock? (A, B, or C): ").strip().upper()

    time.sleep(0.5)
    if choice == "B":
        print("\n✨ *CLICK!* The silver key turns smoothly!")
        print("🚪 The massive stone door slowly grinds open...")
        print("💎 An ancient vault overflowing with glowing gold and gems is revealed!")
        print("🏆 YOU WIN! Treasure Unlocked! 🗝️✨")
    else:
        print("\n❌ *SKRRRCH!* The key jams and fails to turn.")
        print("🔒 The magical trap locks the chest forever!")
        print("💀 GAME OVER! Wrong key chosen.")

def main():
    while True:
        banner("Choice Board Showcase Menu")
        print("1. Play Theme B: Traffic Dodge")
        print("2. Play Theme E: Treasure Door")
        print("3. View Prompts for All 5 Themes")
        print("4. Exit")

        c = input("\nEnter choice (1-4): ").strip()
        if c == "1":
            play_traffic_dodge()
        elif c == "2":
            play_treasure_door()
        elif c == "3":
            banner("The 5 Choice Board Themes")
            print("Theme A: Fruit Catcher 🍎 (Move basket left/right to catch apples)")
            print("Theme B: Traffic Dodge 🏎️ (Steer car to dodge cones on highway)")
            print("Theme C: Coral Explorer 🤿 (Diver swims 4-way to collect pearls)")
            print("Theme D: Jungle Runner 🐒 (Monkey auto-runs and jumps over logs)")
            print("Theme E: Treasure Door 🚪 (Click the correct key to unlock castle door)")
        elif c == "4":
            print("\nCongratulations on becoming a certified Prompt Engineer! 🎓🚀\n")
            break
        else:
            print("Invalid selection.")
        input("\nPress Enter to return to menu...")

if __name__ == "__main__":
    main()
